macos/sol-macOS/lib/watchPasteboard.swift

```diff
@@ -0,0 +1,218 @@
+    let metaData = try! JSONEncoder().encode(meta)
+    let metaFile = itemDirectory.appendingPathComponent("meta.json")
+    try! FileManager.default.createDirectory(atPath: self.itemDirectory.path, withIntermediateDirectories: true, attributes: nil)
+    try! metaData.write(to: metaFile, options: [])
+  }
+}
+
+struct SavedItemMeta: Codable {
+  var fileExtension: String
+  var hash: String
+  var size: Int
+  var timesCopied: Int
+  var lastCopied: Date
+}
+
+func handlePastedText(_ text: String, fileExtension: String) {
+    // Create hash of content
+  let hash = text.sha256()
+  
+  // Get the existing item or construct a new one
+  let existingItem = clipboardHistory.getItem(hash: hash)
+  var item: ClipboardTextItem = existingItem ?? ClipboardTextItem.init(text: text, itemDirectory: itemDirectory, fileExtension: fileExtension, hash: hash, timesCopied: 0, lastCopied: Date())
+  // Increment the timesCopied and save it
+  item.timesCopied += 1
+  item.lastCopied = Date()
+  item.saveMeta()
+
+  if existingItem != nil {
+    return
+  }
+
+  do {
+    let itemDirectory = getClipboardDirectory().appendingPathComponent(hash)
+    let itemFile = itemDirectory.appendingPathComponent("file.\(fileExtension)")
+    try text.write(to: itemFile, atomically: true, encoding: .utf8)
+    // If the clipboard history size is greater than the history limit prune oldest items first
+    
+  } catch {
+    print(error.localizedDescription)
+  }
+}
+
+
+func getTimestamp() -> String {
+  let formatter = ISO8601DateFormatter()
+  formatter.formatOptions = [.withFullDate,
+                             .withTime,
+                             .withDashSeparatorInDate,
+                             .withColonSeparatorInTime,
+                             .withFractionalSeconds]
+  return formatter.string(from: Date())
+}
+
+func getClipboardDirectory() -> URL {
+  let solDirectory = Bundle.main.resourceURL!.appendingPathComponent("clipboard")
+  // Check if the directory exists, if not create it
+  let isDirectory =  UnsafeMutablePointer<ObjCBool>.allocate(capacity: 1)
+  let fileExists = FileManager.default.fileExists(atPath: solDirectory.path, isDirectory: isDirectory)
+  if !fileExists {
+    try! FileManager.default.createDirectory(atPath: solDirectory.path, withIntermediateDirectories: true, attributes: nil)
+  }
+  
+  // If it's not a directory delete the file and create a directory
+  if !isDirectory.pointee.boolValue {
+    try! FileManager.default.removeItem(at: solDirectory)
+    try! FileManager.default.createDirectory(atPath: solDirectory.path, withIntermediateDirectories: true, attributes: nil)
+  }
+  
+  // Deallocate pointer
+  isDirectory.deallocate()
+  
+  return solDirectory
+}
+
+let clipboardHistory = ClipboardHistoryObservable()
+
+enum ClipboardItem {
+  case text(ClipboardTextItem)
+}
+
+class ClipboardHistoryObservable: ObservableObject {
```

Comment:
I'm going to take parts of this implementation, but stuff like this makes me weary, I do not want to embed SwiftUI into sol...